File: CleverFerretV2/core/network/src/main/java/com/cleverferret/v2/core/network/telemetry/TelemetryEvent.java
package com.cleverferret.v2.core.network.telemetry;

import java.time.Instant;
import java.util.Map;

/**
 * Unified telemetry event schema for outbound integrations.
 */
public record TelemetryEvent(
        String schemaVersion,
        String eventName,
        String integration,
        String operation,
        String requestId,
        Instant timestamp,
        boolean success,
        String errorType,
        String userFacingState,
        Map<String, String> attributes
) {
    public static final String SCHEMA_VERSION = "v1";
}

File: CleverFerretV2/core/network/src/main/java/com/cleverferret/v2/core/network/telemetry/TelemetryEvent.kt
package com.cleverferret.v2.core.network.telemetry

import java.time.Instant

data class TelemetryEvent(
    val schemaVersion: String,
    val eventName: String,
    val integration: String,
    val operation: String,
    val requestId: String,
    val timestamp: Instant,
    val success: Boolean,
    val errorType: String?,
    val userFacingState: String?,
    val attributes: Map<String, String>,
) {
    companion object { const val SCHEMA_VERSION: String = "v1" }
}
